main.py

```diff
@@ -3,16 +3,55 @@
 This NApp creates maintenance windows, allowing the maintenance of network
 devices (switch, link, and interface) without receiving alerts.
 """
-import datetime
+from datetime import datetime, timedelta
 
 import pytz
+from attrs import evolve
+from cattrs import BaseValidationError, Converter
```

Comment:
So the reason I selected `attrs` over `pydantic` comes down to issues I was having early on with the IDs. Originally I wanted to use the ID types we have in core, rather than the string representations. With `pydantic`, I was running into issues with the id type not already being a pydantic type due to their dual string/object nature. With `cattrs`, it seemed simpler to handle this special case, even being able to have separate strategies between the front-end and back-end.

Ultimately, the whole idea fell through, as the apis in other napps already use use the string representations rather than object representations for the ids. So I could convert back to pydantic.

Reply:
@Ktmi, I see. Thanks for the information. Since the idea fell through, and although the IDs ended up extremely nicely in the core when building the objects, and since we're using list of string here, which also simplified and ended-up very nice in the API, let's switch to `pydantic` then all things considered.  Since we don't have a blocker here to use `pydantic` let's stick with it, it'll likely also be less cognitive load not to  learn a new equivalent-ish library and having to keep up with upstream updates for existing core developers and new contributors who might join our team.